patches/blockprotocol+0.0.1.patch

```diff
@@ -25,23 +29,78 @@ index 205630a..1cf433a 100644
    operation: BlockProtocolAggregateOperationInput;
    accountId?: string;
  };
-@@ -116,6 +115,7 @@ export type BlockProtocolUploadFileFunction = {
+@@ -116,6 +109,7 @@ export type BlockProtocolUploadFileFunction = {
      url?: string;
      mediaType: BlockProtocolFileMediaType;
    }): Promise<{
 +    accountId: string;
      entityId: string;
      url: string;
      mediaType: BlockProtocolFileMediaType;
-@@ -139,6 +139,7 @@ export type BlockProtocolEntity = {
+@@ -139,6 +133,7 @@ export type BlockProtocolEntity = {
  };
  
  export type BlockProtocolLink = {
 +  linkId: string;
    sourceEntityId: string;
    destinationEntityId: string;
    destinationEntityVersionId?: string | null;
-@@ -170,8 +171,7 @@ export type BlockProtocolCreateLinksFunction = {
+@@ -153,6 +148,40 @@ export type BlockProtocolLinkGroup = {
+   links: BlockProtocolLink[];
+ };
+ 
++export type BlockProtocolAggregateOperationInput = {
++  entityTypeVersionId?: string | null;
++  pageNumber?: number;
++  itemsPerPage?: number;
++  multiSort?: BlockProtocolMultiSort | null;
++  multiFilter?: BlockProtocolMultiFilter | null;
++  entityTypeId?: string;
++};
++
++export type BlockProtocolAggregateOperationResults = BlockProtocolAggregateOperationInput & {
++  pageCount?: number;
++};
++
++export type BlockProtocolLinkedAggregationOperationInput = BlockProtocolAggregateOperationInput & {
++  entityTypeId: string;
++};
++
++export type BlockProtocolUpdateLinksMutationResults = {
++  sourceAccountId: string;
++  sourceEntityId: string;
++  path: string;
++  operation: BlockProtocolLinkedAggregationOperationInput & {
++    pageCount?: number;
++  };
++}
++
++export type BlockProtocolLinkedAggregation = BlockProtocolUpdateLinksMutationResults & {
++  __typename?: string;
++  results: Array<BlockProtocolEntity>;
++  pageCount?: number;
++  entityTypeId: string;
++}
++
++
+ export type BlockProtocolCreateLinksAction = {
+   sourceAccountId?: string | null;
+   sourceEntityId: string;
+@@ -167,11 +196,21 @@ export type BlockProtocolCreateLinksFunction = {
+   (actions: BlockProtocolCreateLinksAction[]): Promise<BlockProtocolLink[]>;
+ };
+ 
++export type BlockProtocolUpdateLinksAction = {
++  sourceAccountId: string;
++  sourceEntityId: string;
++  path: string;
++  updatedOperation: BlockProtocolLinkedAggregationOperationInput;
```

Comment:
This doesn't match the signature of `updateLinks` as described in the spec (down [here](createLinkedAggregationRaw) a little bit), although I appreciate this is because the spec expects a `linkId` to be able to update a link with - @benwerner01 can you remind me why we don't have `linkId` on linked aggregations to identify them with?

Reply:
Because linked aggregations aren't "links", that means they don't have a `linkId` - their identifier is currently a combination of the source entity (`accountId`, `entityId`, and/or `entityVersionId`) and the aggregation path. I would've expected there to be methods for `createLinkedAggregation`/`updateAggregation`/`deleteAggregation` in the spec, just as we have `createLink`/`updateLink`/`deleteLink`. We could always add a `aggregationId` if you deem it easier to work with.

Reply:
I expect it would easier to use an `aggregationId`/`linkedAggregationId` to identify the records.

The approach the spec currently takes is to have a single `createLink` (etc) function which can handle both aggregations and single entity links, depending on whether an `operation` is supplied or not.

We could split them out into `createLink` and `createLinkedAggregation` (and so on).